packs/rabbitmq/sensors/queues_sensor.py

```diff
@@ -1,42 +1,62 @@
 import json
 import pickle
 
+import eventlet
 import pika
+
 from pika.credentials import PlainCredentials
 
-from st2reactor.sensor.base import PollingSensor
+from st2reactor.sensor.base import Sensor
 
 DESERIALIZATION_FUNCTIONS = {
     'json': json.loads,
     'pickle': pickle.loads
 }
 
 
-class RabbitMQSensor(PollingSensor):
-    def __init__(self, sensor_service, config=None, poll_interval=5):
-        super(RabbitMQSensor, self).__init__(sensor_service=sensor_service, config=config,
-                                             poll_interval=poll_interval)
+class RabbitMQQueueSensor(Sensor):
+    """Sensor which monitors a RabbitMQ queue for new messages
 
-    def cleanup(self):
-        if self.conn:
-            self.conn.close()
+    This is a RabbitMQ Queue sensor i.e. it works on the simplest RabbitMQ
+    messaging model as described in https://www.rabbitmq.com/tutorials/tutorial-one-python.html.
 
-    def setup(self):
-        self.queues = None
-        self.conn = None
-        self.channel = None
+    It is capable of simultaneously consuming from multiple queues. Each message is
+    dispatched to stackstorm as a `rabbitmq.new_message` TriggerInstance.
+    """
+    def __init__(self, sensor_service, config=None):
+        super(RabbitMQQueueSensor, self).__init__(sensor_service=sensor_service, config=config)
 
+        self._logger = self._sensor_service.get_logger(name=self.__class__.__name__)
+        self.host = self._config['sensor_config']['host']
         self.username = self._config['sensor_config']['username']
         self.password = self._config['sensor_config']['password']
-        self.queues = self._config['sensor_config']['queues']
-        self.host = self._config['sensor_config']['host']
-        self.deserialization_method = self._config['sensor_config']['deserialization_method']
+
+        queue_sensor_config = self._config['sensor_config']['rabbitmq_queue_sensor']
+        self.queues = queue_sensor_config['queues']
+        if not isinstance(self.queues, list):
+            self.queues = [self.queues]
+        self.deserialization_method = queue_sensor_config['deserialization_method']
 
         supported_methods = DESERIALIZATION_FUNCTIONS.keys()
         if self.deserialization_method and self.deserialization_method not in supported_methods:
             raise ValueError('Invalid deserialization method specified: %s' %
                              (self.deserialization_method))
 
+        self.conn = None
+        self.channel = None
+
+    def run(self):
+        self._logger.info('Starting to consume messages from RabbitMQ for %s', self.queues)
+        # run in an eventlet in-order to yield correctly
+        gt = eventlet.spawn(self.channel.start_consuming)
+        # wait else the sensor will quit
+        gt.wait()
+
+    def cleanup(self):
+        if self.conn:
+            self.conn.close()
```

Comment:
We probably also want to call `self.gt.kill()` here before closing the connection, right?